File: Coding-Challenges/Two-Sum/TwoSum.py
def twoSum(nums: list[int], target: int):
    dictionary = {}
    for index, number in enumerate(nums):
        if(dictionary.get(target - number) != None):
            return [dictionary.get(target - number), index]
        else:
            dictionary[number] = index

print(twoSum([2, 7, 11, 15], 9))

File: Coding-Challenges/Two-Sum/TwoSum.js
function twoSum(nums, sum){
    let dictionary = {};
    for (let i = 0; i < nums.length; i++){
        if (dictionary[nums[i]] != undefined)
            return [dictionary[nums[i]], i];
        else
            dictionary[sum - nums[i]] = i;
    }
}

console.log(twoSum([2, 7, 11, 15], 9));